### PR TITLE
[RNA][OTA] Adding image support

### PR DESCRIPTION
**Pull request by charlesx2013**
Modify `ResourceDrawableIdHelper` to look first in the cache to see if there are any ota images before looking it up in the resource map.

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java [hunk 1]
```diff
@@ -66,15 +86,166 @@ public int getResourceDrawableId(Context context, @Nullable String name) {
...
+    String fileParentSuffix;
+
+    PhoneDensity(
+      int density,
+      @NonNull String fileParentSuffix
+    ) {
+      this.density = density;
+      this.fileParentSuffix = fileParentSuffix;
+    }
+  }
```
Comment (mrkcsc) [inline ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java:250]: For the sake of keeping our branch easy to rebase I would try to move *as much of this as possible* out of this file and into a stand-alone file that this drags from.

**Comment by mrkcsc** [comment]
Where is this branch coming from, just off master?  Once we are happy with the code we need to cherry pick this into our existing release branch and use that.

File: ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java (hunk 1 above)
Comment (charlesx2013) [reply ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java:250]: good point, ill try to move this over to a new file.

File: ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java [hunk 2]
```diff
@@ -66,12 +71,40 @@ public int getResourceDrawableId(Context context, @Nullable String name) {
     }
   }
 
+  private String sanitizeResourceDrawableId(@NonNull String name) {
+    return name.toLowerCase().replace("-", "_");
```
Comment (mrkcsc) [inline ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java:75]: Can we avoid this by updating our existing usage of "ReactAssetUtils" or w/e to just use the "_" format?  We don't use it that much and it's one less special thing to maintain in the fork.

File: ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java [hunk 3]
```diff
@@ -66,12 +71,40 @@ public int getResourceDrawableId(Context context, @Nullable String name) {
...
+      return null;
+    }
+
+    name = sanitizeResourceDrawableId(name);
+
+    // Checks to see if we have an ota version of the file, otherwise default to normal behavior.
+    File otaFile = ResourceDrawableUtils.getDrawableFileFromCache(context, name);
+    if (otaFile != null) {
+      return Drawable.createFromPath(otaFile.getAbsolutePath());
+    }
```
Comment (mrkcsc) [inline ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java:89]: IMO move this chunk as a static method in `ResourceDrawableUtils` or better yet have something like (pseudocode).

```
static class ImageOTAUtils {

   fun getOTADrawableOrNull()
   fun getOTAUriOrNull()

   class ResourceDrawableUtils {
   }
   enum PhoneDensity {
   } 
}
```

Now there will only be two lines changes in this stock RN file and only a single "new" file in our branch that we can easily delete/update in the future.

**Comment by mrkcsc** [comment]
I think once this is ready ur gonna wanna close this PR and cherry pick your final commit into https://github.com/discord/react-native/commits/0.66.4-discord-1

However Den recently added new publishing steps for this fork so I think well need to wait till Monday to confirm this is the right approach.

File: ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java (hunk 3 above)
Comment (charlesx2013) [reply ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java:89]: don't want to clutter this file for sure, but having three separate files that will never have conflicts is about as bad as having one giant file so i'll split things up for readability

**Pull request merged**